src/WooCommerce.php

```diff
@@ -394,6 +404,16 @@ public static function getProductDetailsHtmlDataAttr(\WC_Product $product, $is_d
         $product_details['variant'] = implode(', ', $selected_attributes);
       }
     }
+    elseif (!isset($product_details['variant'])) {
+      $attributes = $product->get_variation_attributes();
+      $selected_attributes = [];
+      foreach ($attributes as $attribute_name => $options) {
+        $selected_attributes[] = $options;
+      }
+      if ($selected_attributes) {
+        $product_details['variant'] = implode(', ', $selected_attributes);
+      }
+    }
```

Comment:
Same goes here and the block before, I haven't tested this but it might be a better way to avoid duplication:

```suggestion
    if (('variable' === $product->get_type() && $is_detail_view) || $product->get_type() === 'variation') {
      $attributes = $product->get_variation_attributes();
      $selected_attributes = [];
      foreach ($attributes as $attribute_name => $options) {
        $attribute = isset($_REQUEST['attribute_' . sanitize_title($attribute_name)]) ? wc_clean(stripslashes(urldecode($_REQUEST['attribute_' . sanitize_title($attribute_name)]))) : '';
        if ($attribute) {
          $attribute_term_data = get_term_by('slug', $attribute, $attribute_name);
          $selected_attributes[] = $attribute_term_data ? $attribute_term_data->name : $attribute;
        }
      }
      if (!$selected_attributes && $product->get_type() === 'variation') {
        foreach ($attributes as $attribute_name => $options) {
          $selected_attributes[] = $options;
        }
      }
      if ($selected_attributes) {
        $product_details['variant'] = implode(', ', $selected_attributes);
      }
    }
```